executor/benchmark_test.go

```diff
@@ -1981,3 +1983,36 @@ func BenchmarkLimitExec(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkReadLastLinesOfHugeLine(b *testing.B) {
+	// step 1. initial a huge line log file
+	hugeLine := make([]byte, 1024*1024*10)
+	for i := range hugeLine {
+		hugeLine[i] = 'a' + byte(i%26)
+	}
+	fileName := "tidb.log"
+	err := ioutil.WriteFile(fileName, hugeLine, 0644)
+	if err != nil {
+		b.Fatal(err)
+	}
+	file, err := os.OpenFile(fileName, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer func() {
+		file.Close()
+		os.Remove(fileName)
+	}()
+	stat, _ := file.Stat()
+	filesize := stat.Size()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, n, err := readLastLines(context.Background(), file, filesize)
```

Comment:
should simulate the context.Done() signal ?

Reply:
This test only uses to do  benchmark.

Simulate the `context.Done` in test, is a little bit hard, so I only do a manual test for this.

Reply:
ok